go/vt/discovery/healthcheck.go

```diff
@@ -427,44 +427,48 @@ func (hc *HealthCheckImpl) updateHealth(th *TabletHealth, shr *query.StreamHealt
 	// add it to the map by target
 	hc.healthData[targetKey][tabletAlias] = th
 
-	if isMasterUpdate {
-		if len(hc.healthy[targetKey]) == 0 {
-			hc.healthy[targetKey] = append(hc.healthy[targetKey], th)
-		} else {
-			// We already have one up server, see if we
-			// need to replace it.
-			if shr.TabletExternallyReparentedTimestamp < hc.healthy[targetKey][0].MasterTermStartTime {
-				log.Warningf("not marking healthy master %s as Up for %s because its MasterTermStartTime is smaller than the highest known timestamp from previous MASTERs %s: %d < %d ",
-					topoproto.TabletAliasString(shr.TabletAlias),
-					topoproto.KeyspaceShardString(shr.Target.Keyspace, shr.Target.Shard),
-					topoproto.TabletAliasString(hc.healthy[targetKey][0].Tablet.Alias),
-					shr.TabletExternallyReparentedTimestamp,
-					hc.healthy[targetKey][0].MasterTermStartTime)
+	if th.Target.TabletType == topodata.TabletType_MASTER {
+		if isPrimaryUp {
+			if len(hc.healthy[targetKey]) == 0 {
+				hc.healthy[targetKey] = append(hc.healthy[targetKey], th)
 			} else {
-				// Just replace it.
-				hc.healthy[targetKey][0] = th
+				// We already have one up server, see if we
+				// need to replace it.
+				if th.MasterTermStartTime < hc.healthy[targetKey][0].MasterTermStartTime {
+					log.Warningf("not marking healthy master %s as Up for %s because its MasterTermStartTime is smaller than the highest known timestamp from previous MASTERs %s: %d < %d ",
+						topoproto.TabletAliasString(th.Tablet.Alias),
+						topoproto.KeyspaceShardString(th.Target.Keyspace, th.Target.Shard),
+						topoproto.TabletAliasString(hc.healthy[targetKey][0].Tablet.Alias),
+						th.MasterTermStartTime,
+						hc.healthy[targetKey][0].MasterTermStartTime)
+				} else {
+					// Just replace it.
+					hc.healthy[targetKey][0] = th
+				}
 			}
+		} else {
```

Comment:
If this section was the reason you added the MASTER check, it may read better if you explicitly checked for it here. Something like `th.Target.TabletType == topodata.TabletType_MASTER && !isPrimaryUp`. But I'm not sure if that means the same thing.

Reply:
The MASTER check was already there but in the form of a boolean (isMasterUpdate).
I can break this into two separate if blocks instead of an `if .. if .. else` if that is easier to read/understand. What you are suggesting will be equivalent to how it is written today.

Reply:
The concern was the deep nesting which made it non-obvious. A cascading if (or switch) may read better. Try it. If it doesn't improve it, we can keep this as is.

Reply:
Ok, I think the switch made it better.